zerver/lib/actions.py

```diff
@@ -5900,3 +5900,61 @@ def do_delete_realm_export(user_profile: UserProfile, export: RealmAuditLog) ->
     export.extra_data = ujson.dumps(export_data)
     export.save(update_fields=['extra_data'])
     notify_realm_export(user_profile)
+
+def get_topic_messages(
+    user_profile: UserProfile,
+    stream: Stream,
+    topic_name: str
+) -> List[Message]:
+    log_statsd_event('get_topic_messages')
+    messages = UserMessage.objects.filter(
+        user_profile=user_profile
+    )
+
+    recipient = stream.recipient
+    messages = messages.filter(message__recipient=recipient)
+
+    messages = filter_by_topic_name_via_message(
+        query=messages,
+        topic_name=topic_name,
+    )
+
+    return messages
```

Comment:
See the above, but also I think this whole function could be like 2 lines of reasonable code; you possibly should just have the caller call `filter_by_topic_name_via_message`.

Reply:
I was getting this lint error:
```
avoid subject as a var at zerver/lib/actions.py line 5910:
message__subject__iexact=topic_name
```
So I called the method `filter_by_topic_name_via_message`

Reply:
Yeah I just mean it could be:
```
query = UserMessage.objects.filter(user_profile=user_profile, message__recipient=stream.recipient)
return filter_by_topic_name_via_message(query=query, topic_name=topic_name)
```

Which might be better inlined unless we need it for tests.